feat: Add comprehensive AI agent instructions for accessibility devel…

joe-watkins opened the pull request:
Adds instructions for AI agents to adhere to. `AGENTS.md` is the newer industry standard to support multiple IDEs with a single instructions file. AI Agents will pull these instructions into every chat context.

Two key sections: Project-specific rules based the this project and Accessibility rules that AIs should respect.

https://github.com/openai/agents.md
https://code.visualstudio.com/docs/copilot/customization/custom-instructions

silverli commented on AGENTS.md:
```diff
@@ -0,0 +1,353 @@
+## Testing Requirements
+- Write unit tests for all new components
+- Update snapshots when component structure changes intentionally
+- Run tests before committing: `npm test`
```
Run linting: `npm run lint` and `npm run lint:scss`

silverli commented on AGENTS.md:
```diff
@@ -0,0 +1,353 @@
+You are an expert full-stack accessibility developer. MagentaA11y V2 is an accessibility accessibility accpeptance criteria documentation platform built with React and TypeScript. The project provides comprehensive accessibility guidelines, testing procedures, and best practices for web and native applications.
+
+## Core Principles
+1. **Accessibility First**: All code and documentation must follow WCAG guidelines and accessibility best practices
```
Should we specify WCAG 2.2 and the WAI-ARIA version?

joe-watkins replied:
Yes! Brought in versions

silverli commented on AGENTS.md:
````diff
@@ -0,0 +1,353 @@
+```
+- Avoid overuse of sectioning elements and ARIA landmarks
+- Don't add any ARIA attributes unless asked to, rely on web standards first
+- Using generic containers when semantic elements would be more appropriate
````
Do you need a "Don't" or "Avoid" in front of this one?

joe-watkins replied:
Yes and Yes - reworked this

MyLightIsOn commented on AGENTS.md:
````diff
@@ -0,0 +1,353 @@
+```html
+<button>Click me</button>
+```
+- Don't apply role="link" to a Link <a>
````
This <a> tag without an attribute is making the following headings look like links.

jtweb23 commented:
Item #1) Line 252
    - `4.5:1` for normal text (defined as up to and including `18.5px` regular or `14pt` bolded)
Checking on the 1.4.3 Contrast minimum for regular text. Did we mean to say ‘14pt` unbolded ??

joe-watkins commented:
@jtweb23 @MyLightIsOn @silverli We could pull in some some content from these instructions too:
https://github.com/github/awesome-copilot/blob/main/instructions/a11y.instructions.md

@jtweb23 There's some good `<table>` content there we could borrow. 
https://github.com/github/awesome-copilot/blob/main/instructions/a11y.instructions.md#table-and-grid-accessibility-acceptance-criteria

joe-watkins commented on AGENTS.md:
```diff
@@ -0,0 +1,353 @@
+### Documentation (Markdown)
+- All documentation files go in `/public/content/documentation/`
+- Follow the structure: `<section>/<category>/<filename>.md`
+- Use the provided templates: `criteria` or `how-to-test`
```
Added :)

joe-watkins commented on AGENTS.md:
```diff
@@ -0,0 +1,353 @@
+5. Add to navigation if needed using `npm run create-md`
+
+### Updating Documentation
+1. Never manually create markdown files - use the script
```
Added :)

joe-watkins commented on AGENTS.md:
```diff
@@ -0,0 +1,353 @@
+Avoid generating content that reflects implicit bias or outdated patterns. Critically assess accessibility choices and flag uncertain implementations.
+
+- **Verification-Oriented Responses**
+When suggesting accessibility implementations or decisions, include reasoning or references to standards (e.g., WCAG, platform guidelines). If uncertainty exists, the assistant should state this clearly.
```
Yes! Edited

joe-watkins commented:
I've folded in all feedback!

> Item #1) Line 252 - `4.5:1` for normal text (defined as up to and including `18.5px` regular or `14pt` bolded) Checking on the 1.4.3 Contrast minimum for regular text. Did we mean to say ‘14pt` unbolded ??

@jtweb23 I've rewritten this to be a little more readable and just went with px. pt conversions are complex. I also moved it up into it's own little section. 

> Item #2) Had a content idea to consider including **Tables**. Maybe we could start with consideration of **simple data tables** versus **presentation layout tables**: Tables that have structured content which must be linearized by screen readers and similar assistive technology are coded as simple data tables. Simple Data Tables ·      Column headers or Row headers in simple data tables use semantic tags. ·      Data Tables do not use CSS such as CSS Flexbox, CSS Grid, other non-semantic visual positioning attributes to indicate relationships in between table data cells and headers. Presentation tables ·      Tables that are used for layout or presentation use role=”presentation”.

@jtweb23 I've added Tables to the bottom of the file :) Great callout

I think we are good when everyone Resolves there feedback items if happy. We can iterate on this as we get in there and Dev with it. I think especially when it comes to some of the React specific stuff. @silverli @MyLightIsOn 

ps. I was also sure to rebase so merge should be cool.

joe-watkins commented on AGENTS.md:
```diff
@@ -0,0 +1,353 @@
+- Follow the structure: `<section>/<category>/<filename>.md`
+- Use the provided templates: `criteria` or `how-to-test`
+- Use the script: `npm run create-md -- <filename> "<relative-path>" <template-type>`
+- Reference the MARKDOWN_GUIDE.md for formatting standards
```
Added :)

silverli commented on AGENTS.md:
```diff
@@ -0,0 +1,447 @@
+
+## Important Notes
+- Never overwrite existing markdown files without explicit permission
+- The `/src/shared/content.json` is auto-generated but can be used as reference to understand existing content structure
```
hmm this doesn't exist in github anymore.. is there a way to link to the live site content.json?

joe-watkins replied:
@silverli It should exist on the user's machine if they've ever ran the project - it's just not checked into the repo. If the user is leveraging Copilot and the file is in their codebase, just ignored, we should be ok here. I've not had great luck with external URLs in agent custom instructions.

silverli reviewed (approved):
looks great!! all feedback has been incorporated from all reviewers
